test/functional/autocmd/termclose_spec.lua

```diff
@@ -1,46 +1,40 @@
 local helpers = require('test.functional.helpers')(after_each)
-local Screen = require('test.functional.ui.screen')
 
-local clear, execute, feed, nvim, nvim_dir = helpers.clear,
-helpers.execute, helpers.feed, helpers.nvim, helpers.nvim_dir
-local eval, eq = helpers.eval, helpers.eq
+local clear, command, feed, nvim, nvim_dir =
+  helpers.clear, helpers.command, helpers.feed, helpers.nvim, helpers.nvim_dir
+local eval, eq, retry =
+  helpers.eval, helpers.eq, helpers.retry
 
 if helpers.pending_win32(pending) then return end
 
 describe('TermClose event', function()
-  local screen
   before_each(function()
     clear()
     nvim('set_option', 'shell', nvim_dir .. '/shell-test')
     nvim('set_option', 'shellcmdflag', 'EXE')
-    screen = Screen.new(20, 4)
-    screen:attach({rgb=false})
   end)
 
-  it('works as expected', function()
-    execute('autocmd TermClose * echomsg "TermClose works!"')
-    execute('terminal')
-    feed('<c-\\><c-n>')
-    screen:expect([[
-      ^ready $             |
-      [Process exited 0]  |
-                          |
-      TermClose works!    |
-    ]])
+  it('triggers when terminal job ends', function()
+    command('autocmd TermClose * let g:test_termclose = 23')
+    command('terminal')
+    command('call jobstop(b:terminal_job_id)')
+    retry(nil, nil, function()
+      eq(23, eval('g:test_termclose'))
+    end)
```

Comment:
Note that this depends on the fact that `g:test_termclose` will throw an error (rather than test-assertion-failure). `retry()` currently doesn't detect test-assertion-failures, only errors.